core/src/components/picker-column/picker-column.tsx

```diff
@@ -74,21 +75,27 @@ export class PickerColumnCmp implements ComponentInterface {
       onEnd: (ev) => this.onEnd(ev),
     });
     this.gesture.enable();
+    // Options have not been initialized yet
+    // Animation must be disabled through the `noAnimate` flag
+    // Otherwise, the options will will render
+    // at the top of the column and transition down
     this.tmrId = setTimeout(() => {
-      this.noAnimate = false;
+      // After initialization, `refresh()` will be called
+      // At this point, animation will be enabled. The options will
+      // animate as they are being selected.
       this.refresh(true);
     }, 250);
   }
 
   componentDidLoad() {
-    const colEl = this.optsEl;
-    if (colEl) {
-      // DOM READ
-      // We perfom a DOM read over a rendered item, this needs to happen after the first render
-      this.optHeight = colEl.firstElementChild ? colEl.firstElementChild.clientHeight : 0;
-    }
+    this.onDomChange();
+  }
 
-    this.refresh();
+  componentDidUpdate(): void | Promise<void> {
+    // Options have changed since last update.
+    if (this.noAnimate) {
```

Comment:
I wonder if it would be better to use a new, separate variable like `colsChanged` to track this, and set both when the cols change? I worry that if we have something else set `noAnimate` in the future, it could lead to unwanted extra `refresh` calls. The variable name doesn't indicate that it's tied to that behavior.